### DISCUSSION TITLE
(Feature Request) Spaced Repetition

### DISCUSSION THREAD

**Discussion by 9s-l-s9** [TriliumNext/Trilium#6156] [Plugin Ideas]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://www.github.com/zadam/trilium/issues) for a feature request that matches the one I want to file, without success.

### Describe feature

This would put Trilium IMO ahead of Obsidian, Roam and many more.
Could be a simple system. We can take inspiration and the learning algorithm from Anki. There also is RemNote. It is an outliner similar to RoamResearch which transforms every bullet point with a special marker into a card (front: before marker, back: after marker). Maybe you can borrow some concepts from there. Besides RemNote (which is not FOSS) other similar apps for inspiration are [Logseq ](https://discuss.logseq.com/t/spaced-repetition-feature/849)or the [twsr plugin ](https://tadeas.dev/twsr/)for tiddlywiki.

Just an idea but could be useful for learning purposes and provide that notes one takes get reviewed on a regular basis instead of becoming otherwise “forgotten data”.

**Comment by agentydragon** [comment]
I'm pretty confident this should be doable in Trilium with not too much JavaScript scripting, so I think core dev time would be better spent on other features that can't be user-implemented by JS.

(That's why I love Trilium - how easy to extend it is :)

I think for Trilium having one note per card would make more sense, because Trilium doesn't have the outliner structure in the same way the Roam-like family does.

**Comment by hero616** [comment]
这个功能的开发在软件上可能不难实现，核心的东西在于以什么算法来实现一个合理的间隔复习。上面提到了Anki，和它类似的还SuperMemo软件。而且SuperMemo这个软件一直在更新算法。你的这个功能我觉得很可能不会加入Trilium，不过你可以考虑使用SuperMemo。

**Comment by sigaloid** [comment]
I mean, as crazy awesome of a feature as this would be to have, that sort of pushes it closer to a learning tool than something for a personal knowledge base. Not really sure that this is within scope for the project. 

I agree with @agentydragon that this would be better implemented as a js plugin
